packages/trpc/server/routers/viewer/slots.tsx

```diff
@@ -151,12 +155,48 @@ export async function getSchedule(
       },
       users: {
         select: {
-          username: true,
+          // username: true,
           ...availabilityUserSelect,
         },
       },
     },
   });
+
+  const isDynamicBooking = !input.eventTypeId;
+  // For dynamic booking, we need to get and update user credentials, schedule and availability in the eventTypeObject as they're required in the new availability logic
+  const dynamicEventType = getDefaultEvent(input.eventTypeSlug);
+  let dynamicEventTypeObject = dynamicEventType;
+
+  if (isDynamicBooking) {
+    const usernameList = input.usernameList;
+    const users = await ctx.prisma.user.findMany({
+      where: {
+        username: {
+          in: usernameList as string[],
+        },
+      },
+      select: {
+        // username: true,
+        allowDynamicBooking: true,
+        ...availabilityUserSelect,
+      },
+    });
+    const isDynamicAllowed = !users.some((user) => {
+      return !user.allowDynamicBooking;
+    });
+    if (!isDynamicAllowed) {
+      throw new TRPCError({
+        message: "Some of the users in this group do not allow dynamic booking",
+        code: "UNAUTHORIZED",
+      });
+    }
+    dynamicEventTypeObject = Object.assign({}, dynamicEventType, {
+      users,
+    });
+  }
+  const startPrismaEventTypeGet = performance.now();
```

Comment:
I got the feeling this should be on line 119 where it was before, here we're not measuring anything significant

Reply:
Yeah, agreed!